app/src/main/java/org/asteroidos/sync/services/SynchronizationService.java

```diff
@@ -233,40 +249,33 @@ public void onCreate() {
             mNM.createNotificationChannel(notificationChannel);
         }
 
-        mBleMngr = get(getApplication());
-        BleManagerConfig cfg = new BleManagerConfig();
-        cfg.forceBondDialog = true;
-        cfg.taskTimeoutRequestFilter = new TaskTimeoutRequestFilter();
-        cfg.defaultScanFilter = new WatchesFilter();
-        cfg.enableCrashResolver = true;
-        cfg.bondFilter = new BondFilter();
-        cfg.alwaysUseAutoConnect = true;
-        cfg.useLeTransportForBonding = true;
-        if (BuildConfig.DEBUG)
-            cfg.loggingEnabled = true;
-        mBleMngr.setConfig(cfg);
 
         mPrefs = getSharedPreferences(MainActivity.PREFS_NAME, Context.MODE_PRIVATE);
         String defaultDevMacAddr = mPrefs.getString(MainActivity.PREFS_DEFAULT_MAC_ADDR, "");
         String defaultLocalName = mPrefs.getString(MainActivity.PREFS_DEFAULT_LOC_NAME, "");
 
-        if(!defaultDevMacAddr.isEmpty()) {
-            if(!mBleMngr.hasDevice(defaultDevMacAddr))
-                mBleMngr.newDevice(defaultDevMacAddr, defaultLocalName);
+        if (mBleMngr == null) {
+            mBleMngr = new AsteroidBleManager(getApplicationContext(), SynchronizationService.this);
+            mBleMngr.setConnectionObserver(this);
+        }
 
-            mDevice = mBleMngr.getDevice(defaultDevMacAddr);
-            mDevice.setListener_State(SynchronizationService.this);
+        if (!(defaultDevMacAddr.equals(""))) {
+            mDevice = BluetoothAdapter.getDefaultAdapter().getRemoteDevice(defaultDevMacAddr);
+        }
 
-            mWeatherService = new WeatherService(getApplicationContext(), mDevice);
-            mNotificationService = new NotificationService(getApplicationContext(), mDevice);
-            mMediaService = new MediaService(getApplicationContext(), mDevice);
-            mScreenshotService = new ScreenshotService(getApplicationContext(), mDevice);
-            mTimeService = new TimeService(getApplicationContext(), mDevice);
-            silentModeService = new SilentModeService(getApplicationContext());
+        if (nonBleServices.isEmpty())
+            nonBleServices.add(new SilentModeService(getApplicationContext()));
 
-            mDevice.connect();
+        if (bleServices.isEmpty()) {
+            // Register Services
+            new MediaService(getApplicationContext(), this);
```

Comment:
When I read this code, I expect all these services to get garbage collected straight away, I don't directly understand why you'd allocate objects without storing them anyway. Now they are implicitely stored because the constructor does `device.registerBleService(this);` but this is a bit insidious.

I think it would be much cleaner to do the registration here:
```
registerBleService(new MediaService(getApplicationContext(), this));
```

And remove it from each constructor. This would also keep the abstraction boundaries clearer. I expect a constructor to only construct an object